src/dbnode/persist/fs/write.go

```diff
@@ -327,13 +329,41 @@ func (w *writer) Close() error {
 	}
 	// NB(xichen): only write out the checkpoint file if there are no errors
 	// encountered between calling writer.Open() and writer.Close().
-	if err := w.writeCheckpointFile(); err != nil {
+	if err := w.writeCheckpointFile(
+		w.checkpointFilePath,
+		w.digestFdWithDigestContents.Digest().Sum32(),
+		w.digestBuf,
+	); err != nil {
 		w.err = err
 		return err
 	}
 	return nil
 }
 
+func (w *writer) DeferClose(cleanup func()) (persist.DeferredCloser, error) {
```

Comment:
We actually don't need to pass in a cleanup function here.

Writing the checkpoint file doesn't require keeping IDs or tags in memory, all the other files are written by the time we only have the checkpoint file to write left.